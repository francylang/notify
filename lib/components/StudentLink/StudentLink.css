.student-card {
  width: 200px;
  display: flex;
  flex-direction: column;
  margin-top: 250px;
  padding: 10px;
  box-shadow: 5px 10px 10px #888888;
  background-color: white;
  border-radius: 20%; }

.student-name {
  text-decoration: none; }
