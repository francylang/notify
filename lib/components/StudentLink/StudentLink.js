import React from 'react';
import PropTypes from 'prop-types';
import '../../index.css';
import { Link } from 'react-router-dom';

const StudentLink = ({ id, name, deleteStudent }) => {

  return (
    <div className='student-card'>
      <div
        className='delete-student-btn'
        onClick={() => deleteStudent(id)}>
        x
      </div>
      <Link
        to={`/students/${id}`}
        className='student-name'>
        {name}
        {}
      </Link>
    </div>
  )
}

StudentLink.propTypes = {
};

export default StudentLink;
