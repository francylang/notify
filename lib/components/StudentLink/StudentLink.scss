$main-color: #1a90bf ;
$main-color-hover: #405780;

.student-card {
  align-items: center;
  width: 179px;
  position: relative;
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 14px;
  padding: 60px 10px;
  box-shadow: 2px 1px 12px #888888;
  border-radius: 8%;
}

.student-name {
  text-decoration: none;
  color: #888888;
  text-align: center;
  font-size: 20px;
}

.student-name:hover {
  font-weight: bold;
}

.student-name:focus {
  outline-color: $main-color;
}

.delete-student-btn {
  color: #888888;
  font-size: 20px;
  width: 10px;
  height: 10px;
  position: absolute;
  top: 10px;
  right: 21px;
  cursor: pointer;
}

.delete-student-btn:hover {
  fill: $main-color;
}

@media screen and (max-width: 320px) {
  .student-card {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin: 12px 0;
    padding: 60px 10px;
    box-shadow: 2px 1px 12px #888888;
    }
  }

@media screen and (max-width: 650px) {
  .student-card {
    width: 62%;
    font-size: 40px;
    border-radius: 8%;
  }

  .student-name {
    font-size: 30px;
  }
}

@media screen and (max-width: 750px) {
  .student-card {
    width: 260px;
  }
}

@media screen and (min-width: 1900px) {
  .student-card {
    width: 200px;
    padding: 80px 10px;
  }

  .teacher-name {
    margin-left: 105px
  }

}
