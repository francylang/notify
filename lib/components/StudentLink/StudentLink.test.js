import React from 'react';
import StudentLink from './StudentLink.js';
import { shallow } from 'enzyme';

describe('<StudentLink />', () => {

  it('should exist', () => {
    const wrapper = shallow(<StudentLink />);

    expect(wrapper).toBeDefined();
  });

  it('should render a student card', () => {
    const wrapper = shallow(<StudentLink />);

    const studentCard = wrapper.find('.student-card');

    expect(studentCard.length).toEqual(1);
  });

  it('should render an icon', () => {
    const wrapper = shallow(<StudentLink />);

    const studentCard = wrapper.find('.delete-icon');

    expect(studentCard.length).toEqual(1);
  });

  it('should match the StudentLink snapshot', () => {
    let wrapper = shallow(<StudentLink />);

    expect(wrapper).toMatchSnapshot();
  });
});
