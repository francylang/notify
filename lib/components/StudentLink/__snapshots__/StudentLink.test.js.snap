// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentLink /> should match the StudentLink snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="student-card"
>
    <div
        className="delete-student-btn"
        onClick={[Function]}
    >
        <img
            className="delete-icon"
            height="20px"
            src="../lib/assets/close-circular-button-of-a-cross.svg"
            width="20px"
        />
    </div>
    <Link
        className="student-name"
        replace={false}
        to="/students/undefined"
    />
</div>,
  "nodes": Array [
    <div
      className="student-card"
>
      <div
            className="delete-student-btn"
            onClick={[Function]}
      >
            <img
                  className="delete-icon"
                  height="20px"
                  src="../lib/assets/close-circular-button-of-a-cross.svg"
                  width="20px"
            />
      </div>
      <Link
            className="student-name"
            replace={false}
            to="/students/undefined"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <StudentLink />,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="student-card"
>
          <div
                    className="delete-student-btn"
                    onClick={[Function]}
          >
                    <img
                              className="delete-icon"
                              height="20px"
                              src="../lib/assets/close-circular-button-of-a-cross.svg"
                              width="20px"
                    />
          </div>
          <Link
                    className="student-name"
                    replace={false}
                    to="/students/undefined"
          />
</div>,
        "_debugID": 8,
        "_renderedOutput": <div
          className="student-card"
>
          <div
                    className="delete-student-btn"
                    onClick={[Function]}
          >
                    <img
                              className="delete-icon"
                              height="20px"
                              src="../lib/assets/close-circular-button-of-a-cross.svg"
                              width="20px"
                    />
          </div>
          <Link
                    className="student-name"
                    replace={false}
                    to="/students/undefined"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <StudentLink />,
}
`;
